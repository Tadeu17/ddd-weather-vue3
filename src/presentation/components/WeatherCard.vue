<template>
  <div class="bg-gray-50 p-4 rounded shadow-md border">
    <h3 class="text-xl font-bold text-gray-900">
      <span>{{ weather.location.name }}, {{ weather.location.country }}</span>
    </h3>
    <p class="text-gray-700">Temperature: <strong>{{ weather.temperature }}Â°C</strong></p>
    <p class="text-gray-700">Humidity: <strong>{{ weather.humidity }}%</strong></p>
    <p class="text-gray-700">Wind Speed: <strong>{{ weather.windSpeed }} km/h</strong></p>
    <p class="text-gray-700">Condition: <strong>{{ weather.description }}</strong></p>
    <p class="text-gray-600 text-sm">Date: {{ formattedDate }}</p>
  </div>
</template>

<script setup lang="ts">
import { computed } from "vue";
import { formatDate } from "@/utils/dateUtils";

import type { WeatherProps } from "./WeatherCard.types";

const props = defineProps<WeatherProps>();

const formattedDate = computed(() => formatDate(props.weather.date));
</script>
