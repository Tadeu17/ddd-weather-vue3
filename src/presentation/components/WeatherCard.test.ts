import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import WeatherCard from "./WeatherCard.vue";
import type { Weather } from "@/domain/entities/Weather";

const mockWeather: Weather = {
  temperature: 25,
  windSpeed: 10,
  description: "Sunny",
  date: new Date("2025-04-04T00:00:00Z"),
};

describe("WeatherCard", () => {
  it("should render weather data correctly", () => {
    const wrapper = mount(WeatherCard, {
      props: { weather: mockWeather },
    });

    expect(wrapper.find("h3 strong").text()).toBe("April 4, 2025");
    expect(wrapper.text()).toContain("25Â°C");
    expect(wrapper.text()).toContain("10 km/h");
    expect(wrapper.text()).toContain("Sunny");
  });
});
