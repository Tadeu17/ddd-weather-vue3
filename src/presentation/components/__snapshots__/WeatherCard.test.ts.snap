// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WeatherCard Component > should match snapshot 1`] = `
"<article class="p-4 rounded shadow-md border">
  <h3 class="text-xl"><strong>Mocked Date</strong></h3>
  <p class="text-gray-700">Temperature: <strong>25Â°C</strong></p>
  <p class="text-gray-700">Wind Speed: <strong>10 km/h</strong></p>
  <p class="text-gray-700">Condition: <strong>Clear sky</strong></p>
</article>"
`;
